### PR TITLE
Added support for never-ending games

### PR DESCRIPTION
**Pull request by ericyc16**
Resolves issue #708.

This pull request updates the game over mechanism for a chiventure game. A chiventure game should end when one of three cases happens:
1. A final room is entered and all end conditions are met
2. No end conditions exist, but a final room is entered
3. No final room exists, but all end conditions are met

If no final room or no end conditions were defined, a game ended after any single action occurs. Pull #698 noticed this and suggested that if no final room is defined and if no end conditions exist, a game should continue indefinitely (or "never end").

Code has been updated so that _only_ the three cases listed above result in a game over. Never-ending games are now possible. To support this, I have made the following changes:
- Change `end_conditions_met` to return false when a game contains no end conditions
- Refactored conditional statements in `actionmanagement.c` to a new function in `game.c`: `is_game_over`, which checks each of the three cases listed above
- Wrote tests to confirm that this update works as expected

### CODE REVIEW

File: tests/game-state/test_game.c [hunk 1]
```diff
@@ -335,6 +335,82 @@ Test(game_end_condition, end_conditions_met)
...
+    expected.str_val = "Valid_Value";
+    unexpected.str_val = "Invalid_Value";
+    
+    item_t *test_item = item_new("test_item", 
+    "test item for is_game_over",
+    "item for testing is_game_over()");
+    set_str_attr(test_item, "Test_Attribute", unexpected.str_val);
+    add_item_to_game(game, test_item);
+    
+    game_action_condition_t *condition = malloc(sizeof(game_action_condition_t));
```
[the remark this answers is not in the document]
Comment (ericyc16) [reply tests/game-state/test_game.c:356]: I don't see a function attribute_condition_new, but I do see a condition_new function. Do you mean that?

[the remark this answers is not in the document]
Comment (ericyc16) [reply tests/game-state/test_game.c]: Ends up that is what he meant, so I made that change

[the remark this answers is not in the document]
Comment (Dumplingz) [reply tests/game-state/test_game.c]: Yep, I realized I had my own branch pulled where I refactored this function. My bad!

File: src/game-state/src/game.c [hunk 2]
```diff
@@ -143,6 +143,20 @@ bool end_conditions_met(game_t *game)
     return true; // all conditions met
 }
 
+/* See game.h */
+bool is_game_over(game_t *game)
+{
+    bool end_case1, end_case2, end_case3;
+
```
Comment (MaxineK36) [inline src/game-state/src/game.c:150]: Can you add some quick comments here labeling each end_case?
i.e. `/* end_case1: A final room and end conditions both exist */

File: tests/game-state/test_game.c [hunk 3]
```diff
@@ -335,6 +335,81 @@ Test(game_end_condition, end_conditions_met)
     cr_assert_eq(test_5, true, "end_conditions_met() does not return true when all end conditions are met");
 }
```
Comment (MaxineK36) [inline tests/game-state/test_game.c:337]: This should be refactored slightly; you should make a helper to do the initial setup, and then make each case its own test.

**Comment by ericyc16** [comment]
After refactoring the `is_game_over` tests and fixing a few style issues, the code updating chiventure's game-over mechanism has been merged with the dev branch. The issue related to this pull has been closed.

At the suggestion of Borja, I will be creating a backlog issue for making the game-over message customizable (right now it’s just `Moved into %s. This is the final room, you've won the game! Press ctrl+D to quit. `, which is quite boring)

**Comment by MaxineK36** [comment]
Issue Score: ✔️++

Comments:
Great work! The summary is particularly clear and easy-to-follow.

**Pull request merged**